src/crypto/CHIPCryptoPAL.cpp

```diff
@@ -608,5 +608,54 @@ CHIP_ERROR EcdsaAsn1SignatureToRaw(size_t fe_length_bytes, const ByteSpan & asn1
     return CHIP_NO_ERROR;
 }
 
+CHIP_ERROR GenerateCompressedFabricId(const Crypto::P256PublicKey & root_public_key, uint64_t fabric_id,
+                                      MutableByteSpan & out_compressed_fabric_id)
+{
+    constexpr uint8_t kUncompressedPointMarker = 0x04;
+    if ((root_public_key.Length() != kP256_PublicKey_Length) || (root_public_key.ConstBytes()[0] != kUncompressedPointMarker))
```

Comment:
I am wondering if `P256PublicKey` class should have a method `bool IsUncompressed()`

Reply:
The format by default is uncompressed right now. I think it is used uncompressed uniformly.

Reply:
Done